browser/importer/brave_profile_writer.cc

```diff
@@ -63,3 +68,128 @@ void BraveProfileWriter::UpdateStats(const BraveStats& stats) {
+  rewards_service->SetPublisherAllowVideos(payments->allow_media_publishers);
+  rewards_service->SetPublisherAllowNonVerified(payments->allow_non_verified);
+  rewards_service->SetPublisherMinVisitTime(payments->min_visit_time);
+  rewards_service->SetPublisherMinVisits(payments->min_visits);
+
+  // Set the excluded sites
+  for (const auto& publisher_key : ledger_.excluded_publishers) {
+    rewards_service->ExcludePublisher(publisher_key);
+  }
+
+  // Set the recurring tips (formerly known as pinned sites)
+  int sum_of_monthly_tips = 0;
+  pinned_item_count_ = 0;
+  for (const auto& item : ledger_.pinned_publishers) {
+    const auto& publisher_key = item.first;
+    const auto& pin_percentage = item.second;
+    // NOTE: this will truncate (ex: 0.90 would be 0, not 1)
+    const int amount_in_bat = (int)((pin_percentage / 100.0) *
+      ledger_.settings.payments.contribution_amount);
+    if (amount_in_bat > 0) {
+      pinned_item_count_++;
+      sum_of_monthly_tips += amount_in_bat;
+      rewards_service->OnDonate(publisher_key, amount_in_bat, true);
+    }
+  }
+
+  // Adjust monthly contribution budget
+  // Some may have been allocated for recurring tips
+  bool auto_contribute_enabled = payments->enabled;
+  const int minimum_monthly_contribution = 10;
+  new_contribution_amount_ = payments->contribution_amount;
+  if (sum_of_monthly_tips > 0) {
+    new_contribution_amount_ -= sum_of_monthly_tips;
+    // If left over budget is too low, turn off auto-contribute
+    if (new_contribution_amount_ < minimum_monthly_contribution) {
+      LOG(INFO) << "Setting auto-contribute to false.\n"
+        << "Recurring contributions take up " << sum_of_monthly_tips << " of the monthly "
+        << payments->contribution_amount << " budget.\nThis leaves " << new_contribution_amount_
+        << " which is less than the minimum monthly auto-contribute amount ("
+        << minimum_monthly_contribution << ").";
+      auto_contribute_enabled = false;
+      new_contribution_amount_ = minimum_monthly_contribution;
+    }
+  }
+  rewards_service->SetUserChangedContribution();
+  rewards_service->SetContributionAmount(new_contribution_amount_);
+  rewards_service->SetAutoContribute(auto_contribute_enabled);
+}
+
+void BraveProfileWriter::UpdateLedger(const BraveLedger& ledger) {
+  rewards_service_ =
+      brave_rewards::RewardsServiceFactory::GetForProfile(profile_);
+  if (!rewards_service_) {
+    LOG(ERROR) << "Failed to get RewardsService for profile.";
+    bridge_ptr_->Cancel();
+    return;
+  }
+
+  ledger_ = BraveLedger(ledger);
+
+  // If a wallet doesn't exist, we need to create one (in order to call RecoverWallet)
+  if (!rewards_service_->IsWalletCreated()) {
+    rewards_service_->AddObserver(this);
+    LOG(INFO) << "Creating wallet to use for import...";
+    rewards_service_->CreateWallet();
+    return;
+  }
+
+  // Avoid overwriting Brave Rewards wallet if it existed BEFORE import happened
+  if (!ledger_.clobber_wallet) {
```

Comment:
I think here needs to be if wallet and if active wallet (has transactions or balance)

Reply:
I'm a little bit weary of the default behavior destroying a wallet (without asking for user confirmation). Even if there aren't any transactions or balance, the user may not realize this is happening. They could have the address written down somewhere (ex: they were thinking about adding funds)

@rebron the main use-case for this would be folks being auto-imported when Muon installs/upgrades to Brave Core. If a wallet exists and it doesn't have a balance, should we destroy it by default?

Reply:
Here's my recommendation. 

For auto-import, this is irrelevant as the user has never seen a Brave Core wallet, by definition. 

For manual import from Muon: If there's no transactions and no balance in the Brave Core wallet, we should overwrite it (because that's the 99% case) but save the old Brave Core wallet under the hood (make it retrievable from a file in the user's disk in the context of resolving a support problem; no UI). 

I'd prefer that we not add in a prompt that will be hard for most users to make sense of and will not be relevant in the vast majority of cases. @rebron @bsclifton